### PR TITLE
Document the custom grpcio-tools command class

### PR DESCRIPTION
**Pull request by soltanmm**
cc @hsaliak I think you asked about this at some point.

### CODE REVIEW

**Comment by soltanmm** [comment]
Preemption: yes, it renders properly on PyPI.

**Comment by hsaliak** [comment]
LGTM thanks for documenting this.

File: tools/distrib/python/grpcio_tools/README.rst [hunk 1]
```diff
@@ -137,3 +137,22 @@ Given protobuf include directories :code:`$INCLUDE`, an output directory
 ::
 
   $ python -m grpc.tools.protoc -I$INCLUDE --python_out=$OUTPUT --grpc_python_out=$OUTPUT $PROTO_FILES
+
+To use as a build-step in distutils-based projects, you may use the provided
```
Comment (nathanielmanistaatgoogle) [inline tools/distrib/python/grpcio_tools/README.rst]: Is "build-step" a term of art? If not, the hyphen strikes me as spurious.

**Comment by nathanielmanistaatgoogle** [comment]
Luhguhtum.

File: tools/distrib/python/grpcio_tools/README.rst (hunk 1 above)
Comment (soltanmm) [inline tools/distrib/python/grpcio_tools/README.rst]: And like that, it's gone.

**Comment by kpayson64** [comment]
I'm not sure this approach works.  We would need to lazy load an import of ```grpc.tools.command.BuildProtoModules``` because we can't assume any dependencies exist until ```setuptools.setup()``` is called.  However, with this approach we would need to reference the import in our call to ```setuptools.setup()```

**Comment by kpayson64** [comment]
I think we should just not document this.  

If someone follows this approach verbatim, it is almost guaranteed not to work.  Even if they only have grpcio-tools as a setup_dependency and they make it available on their dev/packaging environments, there is still an invalid import statement in setup.py that will cause their distributed packages to fail installation.

**Comment by soltanmm** [comment]
Well, guaranteed not to work for... what? This situation is no different than Cython needing to be imported to use it in `setup.py`. I lean towards us not being greedy.

**Comment by kpayson64** [comment]
But in the case of Cython, we can wrap the import statement in a function and do lazy loading.  I don't see how its possible to wrap ```import grpc.tools.command``` in a function.

**Comment by soltanmm** [comment]
Resolved offline: we'll provide an example workaround in the form of a reference to the `grpcio-health-checking` package.

**Comment by soltanmm** [comment]
Edited to include the approach taken in `grpcio-health-checking`. Once that distribution is registered for the first time this documentation will be valid.

File: tools/distrib/python/grpcio_tools/README.rst [hunk 2]
```diff
@@ -137,3 +137,42 @@ Given protobuf include directories :code:`$INCLUDE`, an output directory
...
+
+To use as a build step in distutils-based projects, you may use the provided
+command class in your :code:`setup.py`:
+
+::
+
+  setuptools.setup(
+    # ...
+    cmdclass={
+      'build_proto_modules': grpc.tools.command.BuildProtoModules,
```
Comment (kpayson64) [inline tools/distrib/python/grpcio_tools/README.rst]: If we leave this part, this needs to become ```BuildPackageProtos```

Comment (soltanmm) [inline tools/distrib/python/grpcio_tools/README.rst]: Fixed.

**Comment by kpayson64** [comment]
LGTM

**Pull request merged**